rasterio/tool.py

```diff
@@ -48,6 +49,71 @@ def stats(source):
     return Stats(numpy.min(arr), numpy.max(arr), numpy.mean(arr))
 
 
+def show_hist(source, bins=10, masked=True, title='Histogram'):
```

Comment:
how about `title=None` in the signature and `title or "Histogram for {0}".format(src.name)` when we call `plt.title()`?